src/components/Modal/index.tsx

```diff
@@ -44,6 +47,11 @@ const Modal: FC<ModalProps> = ({ className, children, trigger, ...rest }) => {
       )}
       <Dialog.Portal>
         <Overlay>
+          {onClose && (
+            <Dialog.Close className={styles.CloseButton} onClick={onClose}>
+              <IconClose />
+            </Dialog.Close>
+          )}
```

Comment:
Radix Dialog has an `onOpenChange` prop already - can we tie our `onClose` into that somehow?

Reply:
I think onClose is more generic for modal. But its up to your approval